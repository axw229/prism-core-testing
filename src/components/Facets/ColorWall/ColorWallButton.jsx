import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import kebabCase from 'lodash/kebabCase'

import { filterByFamily } from '../../../actions/loadColors'

class ColorWallButton extends PureComponent {
  constructor (props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  render () {
    const { family, current, routeCurrent } = this.props
    const isChecked = (kebabCase(family) === routeCurrent || family === current)

    return (
      <React.Fragment>
        <input
          type='radio'
          className='color-wall-button'
          name='selectedFamily'
          id={`family-${family}`}
          onClick={this.handleClick}
          value={family}
          defaultChecked={isChecked}
        />
        <label className='color-wall-label' htmlFor={`family-${family}`}>
          {family}
        </label>
      </React.Fragment>
    )
  }

  handleClick () {
    this.props.filterByFamily(this.props.family)

    window.location.hash = `/active/color-wall/${kebabCase(this.props.family)}`
  }
}

ColorWallButton.propTypes = {
  family: PropTypes.string,
  current: PropTypes.string,
  filterByFamily: PropTypes.func,
  routeCurrent: PropTypes.string
}

const mapDispatchToProps = (dispatch) => {
  return {
    filterByFamily: (family) => {
      dispatch(filterByFamily(family))
    }
  }
}

export default connect(null, mapDispatchToProps)(ColorWallButton)
