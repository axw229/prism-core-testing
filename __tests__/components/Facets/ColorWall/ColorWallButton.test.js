import React from 'react'
import renderer from 'react-test-renderer'

import ColorWallButton from 'src/components/Facets/ColorWall/ColorWallButton'

describe('<ColorWallButton />', () => {
  test('testing true', () => {
    const state = {
      family: 'All'
    }

    let clickFn = jest.fn()

    const component = renderer.create(
      <ColorWallButton family={state.family} selectFamily={clickFn} />
    ).toJSON()

    expect(component).toMatchSnapshot()

    component[0].props.onChange()

    expect(clickFn).toHaveBeenCalledWith(state.family)
  })
})
