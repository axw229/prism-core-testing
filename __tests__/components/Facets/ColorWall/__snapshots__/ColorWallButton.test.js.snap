// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorWallButton /> testing true 1`] = `
Array [
  <input
    className="color-wall-button"
    id="family-All"
    name="selectedFamily"
    onChange={[Function]}
    type="radio"
    value="All"
  />,
  <label
    className="color-wall-label"
    htmlFor="family-All"
  >
    All
  </label>,
]
`;
